src/Serializer/JsonApiSerializer.php

```diff
@@ -586,6 +586,15 @@ private function serializeIncludedObjectsWithCacheKey($includeObjects, $linkedId
             if (!array_key_exists($cacheKey, $linkedIds)) {
                 $serializedData[] = $object;
                 $linkedIds[$cacheKey] = $object;
+            } else {
+                // Merge any missing relationships that may have been requested from alternative instances of this include
+                if (isset($object['relationships'])) {
+                    foreach ($object['relationships'] as $relationName => $relation) {
```

Comment:
I think adding this prior to line 586 would probably do the same. 
```
if (array_key_exists($cacheKey, $linkedIds) && isset($object['relationships])){
   $object['relationships] = array_merge($object['relationships'], $linkedIds[$cacheKey]['relationships']);
}
```

That would avoid the `else` statement, and nested `foreach`. 

I'll take some time this weekend to test that out.